_posts/2015-11-12-Using-OCR-To-Fix-A-Hilarious-Bug.markdown

```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: 'Using OCR To Fix a Hilarious Big'
+date: 2015-11-05T00:00:00.000Z
+comments: false
+categories: [Ruby, open source, OSS, debugging]
+author: matt
+---
+
+For a little while, we would get very strange bug reports. People would complain that artist thumbnails (viewed in several different contexts across the web and our iOS apps) would not be an image of the artist's work, but rather text, which had inexplicably become an actual JPG. This wasn't just text appearing in a `div` that should contain an `img` or something like that, these were actual JPG's that were pictures of text.
+
+We would fix these as they came up, chalking the strangeness up to some relic of an old image processing pipeline, data being migrated, etc.
+
+However, the reports kept coming in. This blog post is about how we diagnosed this actual bug (linked to 'features' of two different pieces of software we use to manage our data- the fix was then trivial), and how we used a simple Ruby script and OCR to help us detect and fix the existing images.
```

Comment:
The parenthetical can be left out or--better--saved for the end as a lesson to never add features.